pkg/services/object/search/service.go

```diff
@@ -26,6 +27,21 @@ type searchClient interface {
 	searchObjects(*execCtx, client.NodeInfo) ([]oid.ID, error)
 }
 
+// Node represents NeoFS storage node that is served by [Service].
+type Node interface {
```

Comment:
finally, i'm willin to get rid of `...Option` and leave explicit prms. `Node` will work well if we leave one NeoFS-specific parameter - then it will include the local `Select` method. Or we can break it down into node components, like `Network` and `LocalObjects`. 1st approach is better from package pov, that's why i chose it

for now I leave it in a transient state for future PRs

Reply:
> finally, i'm willin to get rid of ...Option and leave explicit prms

yes, that is good, but why is it `Node`?

> for now I leave it in a transient state for future PRs

do you plan to rename it?

Reply:
> Node will work well if we leave one NeoFS-specific parameter - then it will include the local Select method. Or we can break it down into node components, like Network and LocalObjects. 1st approach is better from package pov, that's why i chose it

Reply:
i am confused about how much i do not understand what `Node` is. let @roman-khimov decide

Reply:
If we're to omit `IsOwnPublicKey` then it's a `NodeIterator` of some kind. If not then `Node` is the only thing in common.

Reply:
renamed to `Containers`